src/MPQEditor/MPQEditor.ts

```diff
@@ -0,0 +1,530 @@
+    const provider = new MPQEditorProvider(context);
+    const registrations = [
+      vscode.window.registerCustomEditorProvider(
+        MPQEditorProvider.viewType,
+        provider,
+        {
+          webviewOptions: { retainContextWhenHidden: true },
+        }
+      ),
+      // Add command registration here
+      vscode.commands.registerCommand(
+        "one.editor.mpq.showFromDefaultExplorer",
+        (uri) => {
+          MPQEditorProvider.createMPQJson(uri);
+        }
+      ),
+    ];
+
+    registrations.forEach((disposable) =>
+      context.subscriptions.push(disposable)
+    );
+  }
+
+  /**
+   * find unoccupied mpq file-name
+   */
+  public static findMPQName(baseMPQName: string, dirPath: string): string {
+    const maxIters = 5;
+    for (let i = 0; i < maxIters; i++) {
+      const mpqPath: string = path.join(
+        dirPath,
+        baseMPQName + MPQEditorProvider.fileExtension
+      );
+      if (!fs.existsSync(mpqPath)) {
+        return baseMPQName + MPQEditorProvider.fileExtension;
+      }
+      baseMPQName = baseMPQName + "(1)";
+    }
+    return "";
+  }
+
+  public static async createDefaultMPQ(
+    mpqName: string,
+    dirPath: string,
+    circleName: string
+  ): Promise<vscode.Uri | undefined> {
+    const content =
+      '{"default_quantization_dtype": "uint8",' +
+      '"default_granularity": "channel",' +
+      '"layers": [],' +
+      '"model_path": ' +
+      '"' +
+      circleName +
+      '"' +
+      "}";
+
+    // 'uri' path is not occupied, assured by validateInputPath
+    const uri = vscode.Uri.file(`${dirPath}/${mpqName}`);
+
+    const edit = new vscode.WorkspaceEdit();
+    edit.createFile(uri);
+    edit.insert(uri, new vscode.Position(0, 0), content);
+
+    try {
+      await vscode.workspace.applyEdit(edit);
+      let document = await vscode.workspace.openTextDocument(uri);
+      document.save();
+    } catch (error) {
+      return undefined;
+    }
+
+    return uri;
+  }
+
+  /**
+   * Whether mpqName is valid to to use
+   */
+  public static validateMPQName(
+    dirPath: string,
+    mpqName: string
+  ): string | undefined {
+    const mpqPath: string = path.join(dirPath, mpqName);
+
+    if (!mpqPath.endsWith(MPQEditorProvider.fileExtension)) {
+      return "A file extension must be " + MPQEditorProvider.fileExtension;
+    }
+
+    if (fs.existsSync(mpqPath)) {
+      return `A file or folder ${mpqPath} already exists at this location. Please choose a different name.`;
+    }
+  }
+
+  /**
+   * Create and open for edit default *.mpq.json
+   */
+  public static createMPQJson(uri: vscode.Uri): void {
+    const dirPath = path.parse(uri.path).dir;
+    const modelName = path.parse(uri.path).name;
+    const extName = path.parse(uri.path).ext.slice(1);
+    const circleName = path.parse(uri.path).base;
+
+    // try to guess unoccupied name for mpq.json
+    let mpqName = MPQEditorProvider.findMPQName(modelName, dirPath);
+    if (mpqName.length < 1) {
```

Comment:
```suggestion
    if (mpqName.length == 0) {
```

I think it's more explicit to test an empty string.